notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clf_Ca_final.fit(X_important_train,y_train.values.ravel())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Take a look at our R-squared value on our test data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clf_Ca_final.score(X_important_test,y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's compute the deviance at each stage (n_estimators total stages) for the training and test sets.  Each stage does appear to improve performance!  This seems true with most of our targets, with the exception of Phosphorus (P), which as of this writing has been difficult to fit with trees."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# compute test set deviance\n",
+    "test_score = np.zeros((params['n_estimators'],), dtype=np.float64)\n",
+    "\n",
+    "for i, y_pred in enumerate(clf_Ca_final.staged_predict(X_important_test)):\n",
+    "    test_score[i] = clf_Ca_final.loss_(y_test.values.ravel(), y_pred)\n",
+    "\n",
+    "plt.figure(figsize=(12, 6))\n",
+    "plt.subplot(1, 2, 1)\n",
+    "plt.title('Deviance')\n",
+    "plt.plot(np.arange(params['n_estimators']) + 1, clf_Ca_final.train_score_, 'b-',\n",
+    "         label='Training Set Deviance')\n",
+    "plt.plot(np.arange(params['n_estimators']) + 1, test_score, 'r-',\n",
+    "         label='Test Set Deviance')\n",
+    "plt.legend(loc='upper right')\n",
+    "plt.xlabel('Boosting Iterations')\n",
+    "plt.ylabel('Deviance')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_score"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## RandomizedSearch cross-validation: when there's not enough time for GridSearch\n",
+    "We used the feature_selection package to pare down our feature list to some important ones.  \n",
+    "Now the **model_selection** package will help us to explore the hyperparameter space \n",
+    "and pin down the right combination of hyperparameters that gives us the best model.  \n",
+    "\n",
+    "**GridSearchCV** loops through every combination of user-supplied lists of hyperparameters - \n",
+    "with more than a few parameters, and more than a few values tried for each parameter, \n",
+    "this can compound into a lot of iterations.  \n",
+    "\n",
+    "**RandomizedSearchCV** only calculates for randomly selected parameter sets, \n",
+    "selected from user-supplied distribution functions (here listed in param_dist).  \n",
+    "It can save a lot of time without too much degradation in performance. \n",
```

Comment:
I think it's important to let folks know that RandomizedSearch is actually really good! You could make your statement here stronger and link to the relevant portion of the scikitlearn docs, where there's an excellent explanation.